### ISSUE TITLE
[Bug Report][3.3.2] Keyboard navigation behavior on selects

### ISSUE THREAD

**Issue by paul-thebaud** [vuetifyjs/vuetify#17510]
### Environment
**Vuetify Version:** 3.3.2
**Vue Version:** 3.3.4
**Browsers:** Chrome 113.0.0.0
**OS:** Linux x86_64

### Steps to reproduce
- Open playground
- Focus the first select with tab key
- Open select with arrow down key
- Press tab or shift+tab

### Expected Behavior
The next input (tab) or the previous button (shift+tab) should be focused.

### Actual Behavior
First (on tab) or last (on shift+tab) document's focusable element is focused.

### Reproduction Link
[https://play.vuetifyjs.com/#...](https://play.vuetifyjs.com/#eNqlULFOwzAQ/RXLS5ckVlOmKFRlZIat7pAmhzAkjuW7BqGq/87ZCShtB4rY/N753rv3tkf54Fw2HEAWsiToXFsRrLUVohzSyrn4jKDuLVXGgp+oSO7Jrp8BSSxLNaLZEKGFmkRb7aG91/JphEstRWHYCpnbLh75JZaLRIyvfLHjufpdJv+rzM+p+fmpAZ1nC8wYvVSzShhi7Y0jgUAHty7VCHkkE2k613tKu8plb9hbrvMYtPQ0QC0LEZnAcd8Ba/lK5LBQqm4srzXQmsFnFkhZ16kNf1P+YMl0kDZ9t1llq+xONQZpTmeAXbr3/QeCZxEtk5kNmZfPG6ymn9EhHx0mKuXeMVhcSSv2G8CnHmwDHvytiS7W5qkuRlfJgvtJ2xP33Rr7jhdV1xhr3n4f+a/IUSwo7YLhKZHxTnaOC3L3BZc/Dj0=)

### Other comments
This is due to the menu not being `attached`. When focus is on a menu item, pressing tab will focus next/prev item from the menu position in DOM.
This may be related to #16405 and #16739 (but those are probably outdated since there were a lot of change on selects and lists since).
